torch/_decomp/decompositions.py

```diff
@@ -3241,6 +3241,77 @@ def upsample_bilinear2d(
     return result
 
 
+@register_decomposition(aten.replication_pad2d.default)
+@aten.replication_pad2d.default.py_impl(DispatchKey.Autograd)
```

Comment:
Depending on performance this could be really bad because it forces the operation to always be decomposed, since it happens at the dispatcher level rather than going through the decomposition table. For your purposes I don't think you actually need this though, because the deterministic path doesn't go through the dispatcher.